Do not define _XOPEN_SOURCE in parse.c.

It masks the declaration of crypt in current glibc.

Submitted upstream:

  <http://lists.askja.de/pipermail/ldapvi/2022-December/000128.html>

diff --git a/parse.c b/parse.c
index 460979be468a3c09..a6f777fed00363a6 100644
--- a/parse.c
+++ b/parse.c
@@ -16,7 +16,6 @@
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
  * 02110-1301 USA.
  */
-#define _XOPEN_SOURCE
 #include <unistd.h>
 #include "common.h"
 
